pkg/koordlet/statesinformer/states_noderesourcetopology.go

```diff
@@ -327,24 +327,41 @@ func (s *nodeTopoInformer) reportNodeTopology() {
 		for k, v := range nodeTopoAnnotations {
 			nodeResourceTopology.Annotations[k] = v
 		}
-		// TODO only update if necessary
-		s.updateNodeTopo(nodeResourceTopology)
 
-		// do UPDATE
-		if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
-			_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Update(context.TODO(), nodeResourceTopology, metav1.UpdateOptions{})
-			if err != nil {
-				klog.Errorf("failed to update cpu info of node %s, err: %v", node.Name, err)
-				return err
+		if s.isSyncNeeded(nodeResourceTopology.Annotations) {
+			// TODO only update if necessary
+			s.updateNodeTopo(nodeResourceTopology)
+
+			// do UPDATE
+			if features.DefaultKoordletFeatureGate.Enabled(features.NodeTopologyReport) {
+				_, err = s.topologyClient.TopologyV1alpha1().NodeResourceTopologies().Update(context.TODO(), nodeResourceTopology, metav1.UpdateOptions{})
```

Comment:
Apart from the updating, how about using a NodeResourceTopology [informer](https://book-v1.book.kubebuilder.io/beyond_basics/using_client_go_informers.html) (please refer to the implementation of nodeMetricInformer module) to reduce the cost of getting since the nodeResourceTopology may change infrequently?

Reply:
please add a TODO here: `TODO: use a NodeResourceTopology informer`